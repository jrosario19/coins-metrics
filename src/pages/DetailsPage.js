import { useParams } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import React, { useEffect } from 'react';
import { getCoins } from '../redux/CriptoCoins/CriptoCoins';

const DetailsPage = () => {
  let criptocoin;
  const { coin_id: coinId } = useParams();
  const dispatch = useDispatch();
  const criptocoins = useSelector((state) => state.criptocoins);

  useEffect(() => {
    if (!criptocoins.length) {
      dispatch(getCoins());
    }
  }, [dispatch]);

  if (criptocoins[0]) {
    criptocoin = criptocoins.find((criptocoin) => criptocoin.id === coinId);
  }

  return (
    <>

      <h1>ALL CRIPTO COIN STATISTICS</h1>
      <div className="column-container">
        <div className="criptocoin-container-detail">
          <img className="criptocoin-img" src={criptocoin?.image || criptocoins[0]?.image} alt={criptocoin?.name || criptocoins[0]?.name} />
          <div className="column-container">
            <div>
              <span className="details-name">Name:</span>
              <span className="name-data">{criptocoin?.name || criptocoins[0]?.name}</span>
            </div>

            <div>
              <span className="details-name">Symbol:</span>
              <span className="name-data">{criptocoin?.symbol || criptocoins[0]?.symbol}</span>
            </div>

            <div>
              <span className="details-name">Ranking:</span>
              <span className="name-data">{criptocoin?.market_cap_rank || criptocoins[0]?.market_cap_rank}</span>
            </div>
          </div>
        </div>
        <div className="metadata-container">
          <span className="details-name">Price:</span>
          <span className="name-data">{criptocoin?.current_price || criptocoins[0]?.current_price}</span>
          <span className="details-name">Change %:</span>
          <span className={criptocoin?.price_change_percentage_24h > 0 ? 'price-up' : 'price-down'}>{criptocoin?.price_change_percentage_24h || criptocoins[0]?.price_change_percentage_24h}</span>
        </div>
      </div>

    </>

  );
};

export default DetailsPage;
