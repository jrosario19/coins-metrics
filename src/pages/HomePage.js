import React from 'react';
import CriptoCoinsList from '../components/CriptoCoinsList';

const HomePage = () => (
  <CriptoCoinsList />
);

export default HomePage;
