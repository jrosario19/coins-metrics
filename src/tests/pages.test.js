import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import DetailsPage from '../pages/DetailsPage';
import store from '../redux/configureStore';
import HomePage from '../pages/HomePage';
import { it } from 'mocha';



describe('Renders pages precisely', () => {
  it('Renders Homepage  precisely', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <HomePage />
        </Provider>,
      )
      .toJSON();
    render(
      <Provider store={store}>
        <HomePage />
      </Provider>,
    );

    const element = document.querySelector('.criptocoins-list');
    expect(element).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });

  it('Renders Detailspage  precisely', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <DetailsPage />
        </Provider>,
      )
      .toJSON();
    render(
      <Provider store={store}>
        <DetailsPage />
      </Provider>,
    );

    const element = document.querySelector('.criptocoin-container-detail');
    expect(element).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });
});