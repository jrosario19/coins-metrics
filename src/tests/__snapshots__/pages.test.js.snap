// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders pages precisely Renders Detailspage  precisely 1`] = `
Array [
  <h1
    className="details-headline"
  >
    CRIPTO COIN STATISTICS
  </h1>,
  <div
    className="column-container-price"
  >
    <div
      className="criptocoin-container-detail"
    >
      <img
        className="criptocoin-img-details"
      />
      <div
        className="column-container-details"
      >
        <div>
          <span
            className="details-name1"
          >
            Name:
          </span>
          <span
            className="name-data1"
          />
        </div>
        <div>
          <span
            className="details-name1"
          >
            Symbol:
          </span>
          <span
            className="name-data1"
          />
        </div>
        <div>
          <span
            className="details-name1"
          >
            Ranking:
          </span>
          <span
            className="name-data1"
          />
        </div>
      </div>
    </div>
    <div
      className="metadata-container"
    >
      <span
        className="details-name"
      >
        Price:
      </span>
      <span
        className="name-data"
      />
      <span
        className="details-name"
      >
        Change %:
      </span>
      <span
        className="price-down"
      />
    </div>
  </div>,
]
`;

exports[`Renders pages precisely Renders Homepage  precisely 1`] = `
Array [
  <div
    className="container-input"
  >
    <input
      className="input-field"
      onChange={[Function]}
      placeholder="Search Coin"
      type="text"
    />
  </div>,
  <ul
    className="container criptocoins-list"
  />,
]
`;
